### DISCUSSION TITLE
Manytomany relationship search with criteria

### DISCUSSION THREAD

**Discussion by kirsv-ldnet** [symfony/symfony#43467] [Q&A]
If I have two entities that share many-to-many relationship, how would I do repository matching with criteria from the other end of the relationship?

In my case I have class Event and Resource. If I try to get repository from Event-class and I want to fetch all events that have any of certain resources.

I know multiple way that I can do this by query builder, but I'd like to be able to do it from the repository as it seems to be possible.

Currently using Symfony 4.4. The first problem I encountered was having null-value in assoc-variable in vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php line 1685.

**Comment by SimonVanacco** [comment]
Hi ! 
I'm not too good with Many-to-many relationships, but as you can access the Query Builder from the repository, couldn't you just create yourself a new function in your EventRepository like this :

```
public function findByAssociatedResources(array $resources) {
        $qb = $this->createQueryBuilder('e')
            -> [...]

        $query = $qb->getQuery();

        return $query->execute();
}
```
